### PR TITLE
[desktop] make fonts depend on language and platform

### LINKED ISSUE #1909: Japanese kanji are not Japanese
- [x] This is not a feature request (existing functionality does not work, **not** missing functionality).
 I will request features on [forum](https://www.reddit.com/r/tutanota/) or via support.
- [x] I've searched and did not find a similar issue.

**Describe the bug**
When writing in Japanese, kanji are changed. Also comma after kanji is placed in weird position. In previous versions this was working. In mobile and web application this is shown correctly. 

**To Reproduce**
Steps to reproduce the behaviour:
1. Type this string: "背中、"
2. Here, the font is correct.
3. Please see picture below how this looks in desktop version. You will see that the first set (before comma) is different and comma is in the middle of text.

**Expected behavior**
If you save draft and open with web application, it will be displayed correctly.

**Screenshots**
slika

**Desktop (please complete the following information):**
 - OS: windows 10
 - Version 3.69.2

### PR DESCRIPTION
**Pull request by ganthern**
adds a function getFonts() to main-styles.js that returns the
font-family string for the body style.

will add additional fonts depending on the platform and used language.

AppearanceSettingsViewer will now also cause a regeneration of the
style tags and a change of language on the native side when the language
is changed by the user.

fix #1909

### CODE REVIEW

**Comment by ganthern** [comment]
Still WIP because it's missing  a config migration which could be merged with the one in #1928

**Comment by charlag** [comment]
> we can't rely on locale to detect which font would be correct for the user

is it so? I think that's what Electron will (hopefully) do with https://github.com/electron/electron/pull/23247

**Comment by ganthern** [comment]
Well, at the moment it's not reliable, but they seem to be working on it.
Maybe we could use the LanguageViewModel language tag to choose a default font.
Users that sometimes need a chinese font and at other times a japanese one would need to switch languages then instead of changing the font setting.

**Comment by charlag** [comment]
Hm both sound bad but probably we can cover most cases if we base it on a language and we won't have additional confusing setting.

**Comment by ganthern** [comment]
Another problem is that, as far as I can tell from my research, chinese fonts seem to be inconsistent in what they're called on windows at least, which is why SimHei and 黑体 both are in the list even though they're the same font. The default font setting for the electron browser window takes one ([and only one](https://github.com/electron/electron/blob/b77f701aeba49630735b5b04b4835f01f130a616/shell/browser/web_contents_preferences.cc#L463)) font name, which then is used as default for the sans-serif family.
If a system has the font we chose for the language, but written differently, it wouldn't work.

Changing the body selector in main-styles.js would be another option to get maximum coverage.
In known problematic platform/client/language combinations, we could add a list of fonts to the font-families key.

**Comment by charlag** [comment]
Yeah, probably adding fonts to the body is a good idea (esp. because we discard style blocks currently). Won't screw up all fonts, will fall back properly.

File: src/native/SystemApp.js [hunk 1]
```diff
@@ -21,6 +22,12 @@ export function changeColorTheme(theme: string): Promise<void> {
 	return nativeApp.invokeNative(new Request('changeTheme', [theme]))
 }
 
+export function changeSystemLanguage(language: {code: string, languageTag: string}): Promise<void> {
+	return isDesktop()
```
Comment (charlag) [inline src/native/SystemApp.js:26]: Not sure if we need to stub it here if we have implementation for apps but we can leave it like this for now.

**Comment by charlag** [comment]
I was about to merge this but noticed that https://github.com/electron/electron/issues/18829 is fixed. Maybe we can remove the workaround soon.

**Pull request merged**